yaptide/routes/celery_routes.py

```diff
@@ -165,14 +168,36 @@ def delete(user: UserModel):
 
         # The merge_id is canceled first because merge task starts after run simulation tasks are finished/canceled.
         # We don't want it to run accidentally.
-        celery_app.control.revoke(simulation.merge_id, terminate=True, signal="SIGINT")
-        celery_app.control.revoke(celery_ids, terminate=True, signal="SIGINT")
-        update_simulation_state(simulation=simulation, update_dict={"job_state": EntityState.CANCELED.value})
-        for task in tasks:
-            if task.task_state in [EntityState.PENDING.value, EntityState.RUNNING.value]:
-                update_task_state(task=task, update_dict={"task_state": EntityState.CANCELED.value})
-
-        terminate_unfinished_tasks.delay(simulation_id=simulation.id)
+        if to_fetch is True and simulation.sim_type == SimulationType.SHIELDHIT.value and simulation.job_state in (
+                EntityState.RUNNING.value):
+            logging.debug('Cancelling shieldhit with fetching data')
+            command_as_list = ['kill', '--signal', 'SIGINT']
+            for task in tasks:
+                command_as_list.append(str(task.sim_pid))
+            logging.debug(command_as_list)
+            subprocess.run(command_as_list, check=True)
+        elif to_fetch is True and simulation.sim_type == SimulationType.FLUKA.value and simulation.job_state in (
+                EntityState.RUNNING.value):
+            logging.debug('Cancelling fluka with fetching data')
+            file_name = '/rfluka.stop'
+            for task in tasks:
+                entries = os.listdir(task.path_to_sim)
+                target_folder = [
+                    entry for entry in entries
+                    if entry.startswith('fluka_') and os.path.isdir(os.path.join(task.path_to_sim, entry))
+                ][0]
+                target_path = os.path.join(task.path_to_sim, target_folder)
+                with open(target_path + file_name, 'w') as file:
+                    file.write("")
```

Comment:
maybe these lines could be extracted to a separate function, like `stop_fluka_simulation` with necessary parameteres ?